traitsui/qt4/color_trait.py

```diff
@@ -49,10 +49,10 @@ def convert_to_color(object, name, value):
             tup = channels_to_ints(color_table[value])
 
     if isinstance(tup, tuple):
-        if 3 <= len(tup) <= 4:
+        if 3 <= len(tup) <= 4 and all(isinstance(x, int) for x in tup):
             try:
                 color = QtGui.QColor(*tup)
-            except TypeError:
+            except Exception:
                 raise TraitError
```

Comment:
note: a test already exists
```
    def test_bad_tuple_not_int(self):
        with self.assertRaises(TraitError):
            ObjectWithColor(color=("not an int", 0xff, 0xff))
```

Reply:
Yes, it was failure of that test which prompted this PR.

Reply:
Yes, it was failure of that test which prompted this PR.

Reply:
Yes, it was failure of that test which prompted this PR.

Reply:
Yes, it was failure of that test which prompted this PR.